test/e2e/e2e_test.go

```diff
@@ -399,37 +407,9 @@ var _ = Describe("test e2e inter-node connectivity between worker nodes hybrid o
 		framework.ExpectNoError(err)
 
 		// Wait for pod exgw setup to be almost ready
-		wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
-			jsonFlag := "jsonpath='{.metadata.annotations.k8s\\.ovn\\.org/hybrid-overlay-external-gw}'"
-			kubectlOut, err := framework.RunKubectl("get", "pod", ovnNsFlag, dstPingPodName, jsonFlag)
```

Comment:
I remember now why I've removed it, the old code was not doing anything with the error, actually it always failing to check the annotation because `pingTarget` is an empy string, is initialized later :smile:

Reply:
```
wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
```
we don't check for the error, in this case it will always return an error because of the "uninitialized variable".
it seems that this annotations are not used in the pods, just in the namespace-

Anyway, if the e2e test works we can infer that the logic works, for this more functional checks an unit test will be better

Reply:
recovered the previous check with a comment about the issue